### ISSUE TITLE
Mocking fails if proxy is set

### ISSUE THREAD

**Issue by OmegaDroid** [lundberg/respx#52]
When setting a proxy on the client the mocking fails and tries to connect to the proxy. The behaviour can be seen with the following test case:

```
@respx.mock
def test_proxy_set():
    respx.get("https://google.com", content_type="application/json", content={"foo": "bar"})
    with httpx.Client(proxies={"https": "https://1.1.1.1:1"}) as client:
        response = client.get("https://google.com")
    assert response.json() == {"foo": "bar"}
```

I get the following error: 
```
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
../../../.pyenv/versions/3.8.0/envs/respx/lib/python3.8/site-packages/httpx/_client.py:706: in get
    return self.request(
../../../.pyenv/versions/3.8.0/envs/respx/lib/python3.8/site-packages/httpx/_client.py:570: in request
    return self.send(
../../../.pyenv/versions/3.8.0/envs/respx/lib/python3.8/site-packages/respx/mocks.py:156: in unbound_sync_send
    return self.__Client__send__spy(client, request, **kwargs)
../../../.pyenv/versions/3.8.0/envs/respx/lib/python3.8/site-packages/respx/mocks.py:516: in __Client__send__spy
    response = _Client__send(client, request, **kwargs)
../../../.pyenv/versions/3.8.0/envs/respx/lib/python3.8/site-packages/httpx/_client.py:590: in send
    response = self.send_handling_redirects(
../../../.pyenv/versions/3.8.0/envs/respx/lib/python3.8/site-packages/httpx/_client.py:617: in send_handling_redirects
    response = self.send_handling_auth(
../../../.pyenv/versions/3.8.0/envs/respx/lib/python3.8/site-packages/httpx/_client.py:654: in send_handling_auth
    response = self.send_single_request(request, timeout)
../../../.pyenv/versions/3.8.0/envs/respx/lib/python3.8/site-packages/httpx/_client.py:678: in send_single_request
    response = dispatcher.send(request, timeout=timeout)
../../../.pyenv/versions/3.8.0/envs/respx/lib/python3.8/site-packages/httpx/_dispatch/urllib3.py:98: in send
    conn = self.pool.urlopen(
../../../.pyenv/versions/3.8.0/lib/python3.8/contextlib.py:131: in __exit__
    self.gen.throw(type, value, traceback)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

exception_classes = (<class 'urllib3.exceptions.MaxRetryError'>, <class 'urllib3.exceptions.SSLError'>, <class 'OSError'>), cls = <class 'urllib3.exceptions.MaxRetryError'>

    @contextlib.contextmanager
    def as_network_error(*exception_classes: type) -> typing.Iterator[None]:
        try:
            yield
        except BaseException as exc:
            for cls in exception_classes:
                if isinstance(exc, cls):
>                   raise NetworkError(exc) from exc
E                   httpx._exceptions.NetworkError: HTTPSConnectionPool(host='google.com', port=443): Max retries exceeded with url: / (Caused by ConnectTimeoutError(<urllib3.connection.HTTPSConnection object at 0x7ff47c24fca0>, 'Connection to 1.1.1.1 timed out. (connect timeout=5.0)'))

../../../.pyenv/versions/3.8.0/envs/respx/lib/python3.8/site-packages/httpx/_utils.py:368: NetworkError
```

Cheers

**Comment by lundberg** [comment]
Thanks for reporting. Haven't reproduced this yet, but will.

**Comment by lundberg** [comment]
Version `0.11.0` is now released @OmegaDroid, haven't yet tried this, but this new release might solve this, please try it out and report back.

**Comment by lundberg** [comment]
Moving forward....released version `0.11.1` which includes actual proxy handling and test case, so your issue should be solved @OmegaDroid .

**Comment by OmegaDroid** [comment]
Hi @lundberg,

I've just tested my example and it all seems to work as expected on `0.11.1`, thanks a lot.
